### PR TITLE
Add the ability to customize the progress bars

### PR DESCRIPTION
**Pull request by rgreinho**
Gives the user the ability to customize the visual aspects of the main
and/or the child progress bars. The bars can now also be disabled if
needed.

Fixes rgreinho/trauma#5
Fixes rgreinho/trauma#6

### CODE REVIEW

File: src/downloader.rs [hunk 1]
```diff
@@ -225,21 +243,174 @@ impl Default for DownloaderBuilder {
...
+    /// Use fade-in blocks as progress characters: `"█▓▒░  "`.
+    pub const CHARS_FADE_IN: &'static str = "█▓▒░  ";
+    /// Use fine blocks as progress characters: `"█▉▊▋▌▍▎▏  "`.
+    pub const CHARS_FINE: &'static str = "█▉▊▋▌▍▎▏  ";
+    /// Use a line as progress characters: `"━╾─"`.
+    pub const CHARS_LINE: &'static str = "━╾╴─";
+    /// Use rough blocks as progress characters: `"█  "`.
+    pub const CHARS_ROUGH: &'static str = "█  ";
+    /// Use increasing height blocks as progress characters: `"█▇▆▅▄▃▂▁  "`.
+    pub const CHARS_VERTICAL: &'static str = "█▇▆▅▄▃▂▁  ";
```
Comment (mkatychev) [inline src/downloader.rs:346]: looks clean!

File: src/downloader.rs [hunk 2]
```diff
@@ -225,21 +243,174 @@ impl Default for DownloaderBuilder {
...
+        Self {
+            template,
+            progress_chars,
+            enabled,
+            clear,
+        }
+    }
+
+    /// Create a [`ProgressStyle`] based on the provided options.
+    pub fn to_progress_style(self) -> ProgressStyle {
```
Comment (mkatychev) [inline src/downloader.rs:364]: You could probably do this as `impl From<ProgressBarOpts> for ProgressStyle`

Comment (mkatychev) [reply src/downloader.rs:364]:
```diff
- let style = self.to_progress_style(); 
+ let style = self.into(); 
 ```
https://github.com/rgreinho/trauma/blob/8f5b3986785f0963f6018c05bcf48050449147c2/src/downloader.rs#L383

File: src/downloader.rs (hunk 1 above)
Comment (rgreinho) [reply src/downloader.rs:346]: 🤗

File: src/downloader.rs (hunk 2 above)
Comment (rgreinho) [reply src/downloader.rs:364]: I thought of it, but then, I thought that `impl From<T>` implies a conversion. However here we are using some values from the options to build a ProgressStyle. Therefore It did not seem very idiomatic to me.

But maybe I was thinking about it too much. If you still don't see any problem after reading this, I'll implement the 
`From` trait.

**Pull request merged**